File: tests/benchmarks/loop_sum_benchmark.js
#!/usr/bin/env node

const start = process.hrtime.bigint();

let x = 0n;  // Use BigInt for large number precision
for (let i = 0n; i < 1000000000n; i++) {
    x += i;
}

const end = process.hrtime.bigint();
console.log(x.toString());

// Print timing to stderr
const duration = Number(end - start) / 1000000000;  // Convert to seconds
console.error(`Node.js execution time: ${duration.toFixed(6)} seconds`);

File: tests/benchmarks/loop_sum_benchmark.lua
#!/usr/bin/env lua

local start_time = os.clock()

local x = 0
for i = 0, 999999999 do
    x = x + i
end

local end_time = os.clock()
print(x)

-- Print timing info to stderr
io.stderr:write(string.format("Lua execution time: %.6f seconds\n", end_time - start_time))